[dtensor][11/n] adds some __str__ for ease of read

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111278
* #111234
* #110717
* #109145

This add __Str__ to op schema and dtensor spec for ease of reading

wanchaol committed:
[dtensor][11/n] adds some __str__ for ease of read

This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor][11/n] adds some __str__ for ease of read"


This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

fduwjj commented on torch/distributed/_tensor/placement_types.py:
```diff
@@ -419,6 +419,22 @@ def __eq__(self, __o: object) -> bool:
+        if self.tensor_meta is not None:
+            tensor_shape = str(tuple(self.tensor_meta.shape))
+        else:
+            tensor_shape = "?"
```
`?` is a little bit weird? Can we say unknown shape?

fduwjj commented on torch/distributed/_tensor/op_schema.py:
```diff
@@ -207,6 +207,24 @@ def __repr__(self) -> str:
+    def __str__(self) -> str:
+        args_sharding: List[str] = []
+        mesh = None
+        for spec in self.args_schema:
```
nit: maybe just call it `arg` instead of spec because it's not always `spec`?

wanchaol committed:
Update on "[dtensor][11/n] adds some __str__ for ease of read"


This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor][11/n] adds some __str__ for ease of read"


This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor][11/n] adds some __str__ for ease of read"


This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor][11/n] adds some __str__ for ease of read"


This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor][11/n] adds some __str__ for ease of read"


This add __Str__ to op schema and dtensor spec for ease of reading

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
